compiler/rustc_hir_analysis/src/collect/resolve_bound_vars.rs

```diff
@@ -1180,6 +1133,25 @@ impl<'a, 'tcx> BoundVarContext<'a, 'tcx> {
         self.with(scope, walk);
     }
 
+    fn visit_early<F>(&mut self, hir_id: hir::HirId, generics: &'tcx hir::Generics<'tcx>, walk: F)
```

Comment:
I've introduced `visit_early` (aligning with the preexisting `visit_early_late`) to combat duplication. It has 5 call sites.